### DISCUSSION TITLE
Conflicts between NextAuth using pages/api and custom API directory on Vercel deployment

### DISCUSSION THREAD

**Discussion by hassanpasha5630** [vercel/community#4476] [Help]
### Summary

Hello everyone,

I am currently experiencing an issue with my Next.js application deployed on Vercel, specifically related to NextAuth and my custom API directory.

Setup and Issue:

I have set up NextAuth to use pages/api for authentication routes as it's the default and recommended setup.
I also have another custom API directory which is essential for my application's functioning.
Locally, everything works perfectly; NextAuth routes do not clash with my custom API routes. However, when I deploy to Vercel, issues arise, and I believe it's related to how Vercel handles the api at the root directory.
Errors and Observations:

On production, I receive a [next-auth][error][CLIENT_FETCH_ERROR] pointing to an unexpected token which indicates a possible routing or response issue.
Further inspection led me to believe that there's a potential overlap or confusion between Vercel's handling of the api directory and NextAuth's usage of pages/api.
Steps I've Taken:

Tried moving NextAuth routes out of pages/api but faced errors related to Next.js' expectations of named exports corresponding to HTTP verbs.
Updated environment variables and configurations to reflect any changes made to directory structures.
Thorough testing locally to ensure the setup works, but the production deployment still faces the same issues.

Is there a recommended way to make NextAuth work outside of the pages/api directory, especially considering Vercel's special handling of the root api directory?
If not, are there any workarounds or configurations that can help in resolving this clash between NextAuth's pages/api and Vercel's api handling?
I would appreciate any guidance or insights into resolving this. Thank you!

### Steps to Reproduce

Create api at the root and setup nextauth at pages/api

**Comment by amyegan** [comment]
Hi @hassanpasha5630! I don't recommend using both the `/api` and `/pages/api` because it can cause the routing conflicts you saw. Since your main app seems to use Next.js, I would stick with the `/pages/api` directory for NextAuth.

If your custom API functions use the Node.js Runtime, then you can also put that code in the `/pages/api` directory. If you're using a different runtime for your custom API, you could move it to a different project and [add CORS headers](https://vercel.com/guides/how-to-enable-cors) or [use rewrites as a reverse proxy](https://vercel.com/guides/vercel-reverse-proxy-rewrites-external).
